react-app-drawer-component/src/AppDrawer.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
+
+function AppDrawer() {
+  const [open, setOpen] = useState(false);
+
+  function handleClick() {
+    setOpen((prev) => !prev);
```

Comment:
In this case (and in almost all the cases you will encounter), just `setOpen(!open);` works and is cleaner.